src/SQLParser/Query/Union.php

```diff
@@ -99,27 +131,33 @@ public function walk(VisitorInterface $visitor)
         }
         if ($result !== NodeTraverser::DONT_TRAVERSE_CHILDREN) {
             $this->walkChildren($this->selects, $visitor);
+            $this->walkChildren($this->order, $visitor);
         }
 
         return $visitor->leaveNode($node);
     }
 
-    /**
-     * @param array<Select|null> $children
-     * @param VisitorInterface $visitor
-     */
-    private function walkChildren(array &$children, VisitorInterface $visitor): void
+    private function walkChildren(&$children, VisitorInterface $visitor): void
     {
         if ($children) {
-            foreach ($children as $key => $operand) {
-                if ($operand) {
-                    $result2 = $operand->walk($visitor);
-                    if ($result2 === NodeTraverser::REMOVE_NODE) {
-                        unset($children[$key]);
-                    } elseif ($result2 instanceof NodeInterface) {
-                        $children[$key] = $result2;
+            if (is_array($children)) {
+                foreach ($children as $key => $operand) {
+                    if ($operand) {
+                        $result2 = $operand->walk($visitor);
+                        if ($result2 === NodeTraverser::REMOVE_NODE) {
+                            unset($children[$key]);
+                        } elseif ($result2 instanceof NodeInterface) {
+                            $children[$key] = $result2;
+                        }
                     }
                 }
+            } else {
```

Comment:
Because the type of `$order` is `NodeInterface[]|NodeInterface`, I took inspiration from the Select Class for the type. Not sure if it is necessary in this case though.

Since I update the php doc type hint, do I need to add comment in the code ?

Reply:
Okay, there is no comment there so it is fine, thank you.